### PR TITLE
Support for in-band-mgmt via management VRF

### PR DESCRIPTION
**Pull request by venkatmahalingam**
Signed-off-by: Venkatesan Mahalingam <venkatesan_mahalinga@dell.com>

Please refer [HLD](https://github.com/Azure/SONiC/pull/638) and [swss-common](https://github.com/Azure/sonic-swss-common/pull/479) pull requests for more information.

**What I did**

Support for in-band-mgmt via management VRF

**Why I did it**
Other than eth0, mgmt VRF should have in-band L3 interfaces as well.

**How I verified it**
Verified the functionality using sonic-vs testcases

**Details if related**

### CODE REVIEW

**Comment by venkatmahalingam** [comment]
@prsunny Please take a look at the in-band-mgmt via mgmt VRF support changes.

**Comment by venkatmahalingam** [comment]
@prsunny Can you please review this PR?

File: cfgmgr/vrfmgr.cpp [hunk 1]
```diff
@@ -207,6 +225,46 @@ void VrfMgr::doTask(Consumer &consumer)
         auto vrfName = kfvKey(t);
 
         string op = kfvOp(t);
+		// Mgmt VRF table event handling for in-band management
+        if (consumer.getTableName() == CFG_MGMT_VRF_CONFIG_TABLE_NAME) {
```
Comment (prsunny) [inline cfgmgr/vrfmgr.cpp:229]: { -> next line. Please do it for rest of the changes

File: cfgmgr/vrfmgr.cpp [hunk 2]
```diff
@@ -207,6 +225,46 @@ void VrfMgr::doTask(Consumer &consumer)
         auto vrfName = kfvKey(t);
 
         string op = kfvOp(t);
+		// Mgmt VRF table event handling for in-band management
```
Comment (prsunny) [inline cfgmgr/vrfmgr.cpp:228]: Please align

File: cfgmgr/vrfmgr.cpp [hunk 3]
```diff
@@ -207,6 +225,46 @@ void VrfMgr::doTask(Consumer &consumer)
...
+            if (op == SET_COMMAND) {
+                bool in_band_mgmt_enabled = false;
+                bool mgmt_vrf_enabled = false;
+                for (auto i : kfvFieldsValues(t))
+                {
+                    if (fvField(i) == "mgmtVrfEnabled") {
+                        if (fvValue(i) == "true") {
+                            mgmt_vrf_enabled = true;
+                        } else {
+                            op = DEL_COMMAND;
```
Comment (prsunny) [inline cfgmgr/vrfmgr.cpp:240]: Why are we overwriting the `op` command here?. This operation is expected from the notification

File: orchagent/intfsorch.cpp [hunk 4]
```diff
@@ -159,6 +160,19 @@ string IntfsOrch::getRouterIntfsAlias(const IpAddress &ip, const string &vrf_nam
...
 }
 
+bool IntfsOrch::isInbandIntfInMgmtVrf(const string& alias)
+{
+    if (m_syncdIntfses.find(alias) == m_syncdIntfses.end())
+        return false;
+
+    string vrf_name = "";
+    vrf_name = m_syncdIntfses[alias].vrf_name;
+    if ((!vrf_name.empty()) && (vrf_name == MGMT_VRF))
```
Comment (prsunny) [inline orchagent/intfsorch.cpp:170]: open/close brace. Same for other parts of code

File: orchagent/intfsorch.cpp [hunk 5]
```diff
@@ -389,7 +403,8 @@ set<IpPrefix> IntfsOrch:: getSubnetRoutes()
     return subnet_routes;
 }
 
-bool IntfsOrch::setIntf(const string& alias, sai_object_id_t vrf_id, const IpPrefix *ip_prefix, const bool adminUp, const uint32_t mtu)
+bool IntfsOrch::setIntf(const string& alias, const string& vrf_name, sai_object_id_t vrf_id, const IpPrefix *ip_prefix,
```
Comment (prsunny) [inline orchagent/intfsorch.cpp:406]: Lets discuss this - Why we need to pass the vrf_name to this function.

File: cfgmgr/vrfmgr.cpp (hunk 1 above)
Comment (venkatmahalingam) [reply cfgmgr/vrfmgr.cpp]: done

File: cfgmgr/vrfmgr.cpp (hunk 3 above)
Comment (venkatmahalingam) [reply cfgmgr/vrfmgr.cpp:246]: We should delete the VRF (Virtual router) from HW when no in-band-intf is part of the mgmt VRF i.e when in_band_mgmt_enabled is set false, we can set the op as DEL_COMMAND to delete the VRF.

File: cfgmgr/vrfmgr.cpp (hunk 2 above)
Comment (venkatmahalingam) [reply cfgmgr/vrfmgr.cpp]: done

File: orchagent/intfsorch.cpp (hunk 4 above)
Comment (venkatmahalingam) [reply orchagent/intfsorch.cpp:170]: done

File: orchagent/intfsorch.cpp (hunk 5 above)
Comment (venkatmahalingam) [reply orchagent/intfsorch.cpp:408]: This helps to ignore the HW programming of the nbr with in-band-intf which is part of the mgmt VRF.

**Comment by venkatmahalingam** [comment]
@prsunny I created a new PR based on private branch instead of a master "Support for in-band-mgmt via management VRF #1726".

**Pull request closed without merging**